### PR TITLE
NIFI-10644 Add Message Demarcator-style processing in PublishMQTT

### PR DESCRIPTION
**Pull request by nandorsoma**
# Summary

[NIFI-10644](https://issues.apache.org/jira/browse/NIFI-10644)

Similar to ConsumeMQTT, this pr implements Message Demarcator for processing new-line (or other character sequence) delimited data.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Review by turcsanyip — CHANGES_REQUESTED** [review]
@nandorsoma Thanks for extending PublishMQTT with demarcator-style processing!
Tested the new feature and also the record-style processing which was affected in the change.
Found only some minor typos and copy-paste errors.

File: nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/PublishMQTT.java [hunk 1]
```diff
@@ -109,6 +113,15 @@ public class PublishMQTT extends AbstractMQTTProcessor implements MqttCallback {
             .description("The Record Writer to use for serializing Records before publishing them as an MQTT Message.")
             .build();
 
+    public static final PropertyDescriptor MESSAGE_DEMARCATOR = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(BASE_MESSAGE_DEMARCATOR)
+            .description("With this property, you have an option to publish multiple messages from a single FlowFile. "
+                    + "This property allows you to provide a string (interpreted as UTF-8) to use for splitting "
+                    + "the FlowFile content. This is an optional property ; if not provided, and if not defining a "
+                    + "Record Reader/Writer, each message received will result in a single FlowFile. To enter special "
```
[the remark this answers is not in the document]
Comment (nandorsoma) [reply nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/PublishMQTT.java:121]: Oh, I tried to match this description to the processor, but seems like I missed this sentence. Thx.

File: nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/PublishMQTT.java [hunk 2]
```diff
@@ -336,4 +315,110 @@ public void deliveryComplete(String token) {
...
+        public String getSuccessTemplateMessage() {
+            return PROVENANCE_EVENT_DETAILS_ON_RECORDSET_SUCCESS;
+        }
+    }
+
+    class ProcessDemarcatedContentStrategy implements ProcessStrategy {
+
+        static final String PROVENANCE_EVENT_DETAILS_ON_SINGLE_MESSAGE_FAILURE = "Publish failed after %d successfully published messages.";
+        static final String PROVENANCE_EVENT_DETAILS_ON_SINGLE_MESSAGE_RECOVER = "Successfully finished publishing previously failed messages. Total message count: %d";
+        static final String PROVENANCE_EVENT_DETAILS_ON_SINGLE_MESSAGE_SUCCESS = "Successfully published all messages. Total message count: %d";
```
[the remark this answers is not in the document]
Comment (nandorsoma) [reply nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/PublishMQTT.java:387]: I'm not good in naming. Thx, good idea.

**Comment by nandorsoma** [comment]
Hey @turcsanyip! Thank you for your review, I've applied the requested changes!

**Review by turcsanyip — APPROVED** [review]
@nandorsoma Thanks for the review changes!

+1 LGTM
Will merge when the CI build is ready.

**Pull request closed without merging**